radis/lbl/broadening.py

```diff
@@ -248,7 +248,16 @@ def gaussian_FT(w_centered, hwhm):
 
 
 def pressure_broadening_HWHM(
-    airbrd, selbrd, Tdpair, Tdpsel, pressure_atm, mole_fraction, Tgas, Tref
+    airbrd,
```

Comment:
it's good if this function can be independant of a DataFrame; so it can be used & tested independantly from the rest of the RADIS code
(e.g., to examplify how Radis works / or to validate it,  we sometimes have a step-by-step calculations, where people can call the  `pressure_broadening_HWHM` function on simple numpy arrays)

Can you think of a way to implement this ? 

An idea could be to use a dictionary to combine `diluent` and the columns you need from `df` (making the dictionary would be done in Radis main code by the function calling pressure_broadening_HWHM() ) 

The dictionary could even include "self" and "air" (at this point there is no difference between them or other collisioniers)

Default behavior :
```
pressure_broadening_HWHM(pressure_atm, Tgas, Tref, {"self": (mole_fraction, broadening_coefficient[i.e. selfbrd], temperature_dependance_coefficient[i.e. Tdpsel ) })
```
With air : 
```
pressure_broadening_HWHM(pressure_atm, Tgas, Tref, {"self": (mole_fraction, selfbrd, Tdpsel), "air":(mole_fraction, airbrd, Tdpair) })``` 
```

ARbitrary number of species: 
```
pressure_broadening_HWHM(pressure_atm, Tgas, Tref, {"self": (mole_fraction, selfbrd, Tdpsel), "air":(mole_fraction_air, airbrd, Tdpair) }, "species1":(mole_fraction_1, airbrd, Tdpair)``` 
```
Note : 
- you should handle cases where the last coeficnet, ( Tdpair / Tdpsel / Tdp_anyspecies) is None 
- you can add a new warning in this function if mole_fraction_sum != 1 , but don't raise an error or don't suggest anything like using "diluent=..." . Remember this function may be used/called independantly of RADIS 

THis function will become much simpler (therefore easier to test / understand); and the complexity of building the dictionary will be added in the calling RADIS functions.

Reply:
I tried replacing df with a dictionary diluent_broadening_coeff that contains "broadening_coeff" : value  and hence
now pressure_broadening_HWHM  can be independent of df